apache-rat-core/src/it/java/org/apache/rat/ReportTest.java

```diff
@@ -0,0 +1,205 @@
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.filefilter.AbstractFileFilter;
+import org.apache.commons.io.filefilter.DirectoryFileFilter;
+import org.apache.commons.io.filefilter.NameFileFilter;
+import org.apache.commons.io.filefilter.OrFileFilter;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.rat.api.Document;
+import org.apache.rat.api.RatException;
+import org.apache.rat.commandline.Arg;
+import org.apache.rat.document.DocumentName;
+import org.apache.rat.document.DocumentNameMatcher;
+import org.apache.rat.document.FileDocument;
+import org.apache.rat.document.RatDocumentAnalysisException;
+import org.apache.rat.report.RatReport;
+import org.apache.rat.utils.DefaultLog;
+import org.apache.rat.utils.Log;
+import org.apache.rat.walker.DirectoryWalker;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * Integration tests for the {@link Report} class.
+ * Integration tests must have a specific structure.
+ * <ul>
+ *     <li>They are in directories under {@code src/it/resources/ReportText/X} where {@code X} is the name of the test. Usually
+ *     a reference to a reported bug.</li>
+ *     <li>Within the directory is a file named {@code commandLine.txt}. Each line in the file is a single command line
+ *     token. For example "--output-style XML" would be 2 lines in the file "--output-style" and "XML"</li>
+ *     <li>Within the directory there is a subdirectory named {@code src}. This is the root of the file system for RAT to scan.</li>
+ *     <li>There may be a {@code notes.md} describing the test</li>
+ *     <li>There is a {@code verify.groovy} script. When executed:
+ *     <ul>
+ *         <li>The first parameter will be then name of the file that captured the output.</li>
+ *         <li>The second parameter will be the name of the file that captured the log.</li>
+ *         <li>Any assert that fails within the Groovy script will fail the test.</li>
+ *         <li>Any value returned from the script execution will fail the test and the returned value will be
+ *         used as the failure message.</li>
+ *      </ul>
+ *      </li>
+ *      <li>If an exception is expected when Report is run with the command line a file named {@code expected-message.txt}
+ *      must be present in the directory. It must contain text that is expected to be found within the message
+ *      associated with the exception.</li>
+ * </ul>
+ */
+public class ReportTest {
+
+    private String[] asArgs(final List<String> argsList) {
+        return argsList.toArray(new String[argsList.size()]);
+    }
+
+    @ParameterizedTest(name = "{index} {0}")
+    @MethodSource("args")
+    public void integrationTest(String testName, Document commandLineDoc) throws Exception {
```

Comment:
Shouldn't all this parsing functionality be put into the tools-submodule?

Reply:
I guess this testing infra can be used/or extended in the long term to be useful for performance tests as well.